core/sql/optimizer/NATable.cpp

```diff
@@ -86,6 +86,10 @@
 #include "CmpMain.h"
 
 #define MAX_NODE_NAME 9
+#define MAX_PRECISION_ALLOWED  18
+#define HIVE_MAX_PRECISION_ALLOWED  38
+#define MAX_SCALE_ALLOWED     6
```

Comment:
Sorry I misread the SQL reference manual, read again, yes, there is no limitation about Scale.